src/lib.rs

```diff
@@ -0,0 +1,27 @@
+use std::fs;
+use gitignore;
+
+pub fn print_gtree(entries: fs::ReadDir, gitignore: Option<gitignore::File>) -> std::io::Result<()> {
+    for entry in entries {
+        let dir = entry?;
+        let mut ignore = false;
+
+        if let Some(g) = &gitignore {
+            match g.is_excluded(&dir.path()) {
```

Comment:
I'm... not so sure of the quality of this `gitignore` library.  Every invocation of this `is_excluded` method actually walks the entire directory tree below the directory containing the `.gitignore` file which is... bad. Like really truly amazingly bad. It's no good. Don't do it!

https://docs.rs/gitignore/1.0.6/src/gitignore/file.rs.html#48-83

Seriously, the implementation and API of this library is so offensive that I want to fork and basically rewrite it from scratch. :vomiting_face: